cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactTable.java

```diff
@@ -121,6 +126,35 @@ public FactScanner scan(FactScan scan) throws Exception {
                              scan.getStartTs(), scan.getEndTs());
   }
 
+  /**
+   * Delete entries in fact table. {@link FactScan} is used to specify the deletion parameters
+   * @param scan
+   * @throws Exception
+   */
+  public void delete(FactScan scan) throws Exception {
+    byte[] startRow = codec.createStartRowKey(scan.getTagValues(), scan.getMeasureName(), scan.getStartTs());
+    byte[] endRow = codec.createEndRowKey(scan.getTagValues(), scan.getMeasureName(), scan.getEndTs());
+    byte [][] columns = null;
+
+    if (Arrays.equals(startRow, endRow)) {
```

Comment:
Yeah, in fact need this for test in DefaultCube to work, as everything falls in a single row!